src/Templates/ModelTemplate.cshtml

```diff
@@ -156,7 +157,14 @@ func (@(Model.Name.FixedValue.ToShortName()) *@(Model.Name)) UnmarshalJSON(body
         }
             @if ((p.ModelType is CompositeType && (p.ModelType as CompositeTypeGo).IsPolymorphic) || (p.ModelType is EnumType))
             {
+                @if (p.ShouldBeFlattened())
+                {
+        @:@(Model.Name.FixedValue.ToShortName()).@(p.ModelType.Name) = @(CodeNamerGo.Instance.GetVariableName(p.SerializedName))
```

Comment:
It would be good to refactor this to avoid the duplication.

Reply:
Fixed